Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift

```diff
@@ -269,13 +269,25 @@ class SentrySessionReplay: NSObject {
             SentryLog.debug("Could not delete replay segment from disk: \(error.localizedDescription)")
         }
     }
-
+    
     private func convertBreadcrumbs(breadcrumbs: [Breadcrumb], from: Date, until: Date) -> [any SentryRRWebEventProtocol] {
-        return breadcrumbs.filter {
-            guard let time = $0.timestamp, time >= from && time < until else { return false }
-            return true
+        var filteredResult: [Breadcrumb] = []
+        var lastNavigationTime: Date = from.addingTimeInterval(-1)
+        
+        for breadcrumb in breadcrumbs {
```

Comment:
`h`: If I'm not mistaken, we grab the breadcrumbs from the scope. So, if there are too many breadcrumbs on the scope, shouldn't we instead fix it there so events and transactions also benefit from this improvement?

Reply:
This would be some kind of break change. 
Also, this is actually not a bug; it's the way it was designed for Cocoa and view controllers, but not desirable for the session replay player. 

We could discuss if it makes sense to also apply this change to general navigation breadcrumbs. But I would do it in another PR and then remove this change.

Reply:
This is a team decision. The session replay player timeline has too many of the same breadcrumbs in the same place. Breadcrumbs in session replay are used to navigate the replay, but having multiple items at the same point in time doesn’t add any value.

This is a different product with a different presentation. Nobody had complained about navigation breadcrumbs for issues or transactions, let’s not touch it.

Reply:
By the way, perhaps you’re thinking that this PR is removing something that shouldn’t be there or fixing an issue with the source of breadcrumbs. But, that’s not the case. Each individual breadcrumb was created by a distinct ViewController, as a view controller can have multiple child view controllers. This information does make sense for debugging purposes (in issues and transactions).

Reply:
Thanks for explaining. I still don't agree, but I accept the decision.